lib/spack/spack/spec.py

```diff
@@ -4861,6 +4862,46 @@ def reconstruct_virtuals_on_edges(spec):
             edge.update_virtuals([vspec])
 
 
+def format_path(spec, format_string):
```

Comment:
typehints?

Reply:
This moved but I added one. FWIW, even with `format_string: str`, I prefer to call it `format_string`.